src/Downloader.php

```diff
@@ -279,6 +279,18 @@ public function getRecipes(array $operations): array
         return $data;
     }
 
+    /**
+     * Used to "hide" a recipe version so that the next most-recent will be returned.
+     *
+     * This is used when resolving "conflicts".
+     *
+     * @internal
+     */
+    public function removeRecipeFromIndex(string $packageName, string $version)
+    {
+        unset($this->index[$packageName][$version]);
+    }
```

Comment:
By far the cleanest and easiest way to implement this was to: (A) allow the `Downloader::getRecipes()` method to be called normally and then (in the case of a conflict) (B) remove that recipe version from `Downloader` (that's this method) and (C) call `Downloader::getRecipes()` again.